server/routes/api/landing_page.py

```diff
@@ -455,23 +461,31 @@ def data(dcid):
                 del spec_and_stat[category][topic]
             else:
                 spec_and_stat[category][topic] = filtered_charts
-        if not spec_and_stat[category]:
-            del spec_and_stat[category]
+        # Don't delete a category if it is in the valid categories list.
+        if (not spec_and_stat[category]) and ("validCategories"
```

Comment:
done

Reply:
I'm slightly confused on the intentions of when we should be deleting the `spec_and_stat[category]`

Are the following cases possible and are they all cases we would want to delete?
1. not spec_and_stat[category] and category is not in the valid categories list
2. not spec_and_stat[category] and there is no valid categories list

Reply:
so we want to delete category from the spec and stat when spec_and_stat is empty for that category, and the category is not in the valid categories for that place.

So case 1 is basically the case we are looking for. case 2 is a subset of case 1, but that would probably never occur because it essentially means we have no stat var for that place. If that happens, line 420 handles that case by saying that we have no data for that page.